src/Symfony/Component/HttpClient/CurlHttpClient.php

```diff
@@ -274,6 +274,92 @@ public function request(string $method, string $url, array $options = []): Respo
             $curlopts[\CURLOPT_TIMEOUT_MS] = 1000 * $options['max_duration'];
         }
 
+        if (isset($options['extra']['curl']) && \is_array($options['extra']['curl'])) {
+            $curloptsToConfig = [
```

Comment:
thanks, one last note: could you please move the new logic in a separate method in order to ease reading the request() one?

Reply:
done.